capstone-backend/src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -0,0 +1,181 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.sps.data.VolumeData;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.Map;
+import java.util.ArrayList;
+import java.util.Collection;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns search data from the Google Books API. */
+@WebServlet("/api/search")
+public class SearchServlet extends HttpServlet {
+
+  public final static int DEFAULT_MAX_RESULTS = 5;
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String fullOutput = "";
+    String searchTerm = request.getParameter("searchTerm");
+    int maxResults = parseNaturalNumber(request.getParameter("maxResults"));
+    if (maxResults == -1) {
+      maxResults = DEFAULT_MAX_RESULTS;
+    }
+    
+    if (searchTerm == null || searchTerm.isEmpty()) return;
+    searchTerm = URLEncoder.encode(searchTerm, "UTF-8");
+
+    try {
+      String formattedURL = String.format("https://www.googleapis.com/books/v1/volumes?q={%s}&maxResults=%d&country=US",
+        searchTerm, maxResults);
+		  URL url = new URL(formattedURL);
+		  HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+		  conn.setRequestMethod("GET");
+		  conn.setRequestProperty("Accept", "application/json");
+
+      if (conn.getResponseCode() != 200) {
+        throw new RuntimeException("Failed : error code : "
+            + conn.getResponseCode());
+      }
+      BufferedReader br = new BufferedReader(new InputStreamReader(
```

Comment:
I implemented this into my code to read the ErrorStream in the case of a non-200 response code! I wanted to prevent sending any of this error date from going to the frontend, so it gets passed to the System.err.println log instead